src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -2080,4 +2082,44 @@ protected void notifyShutdown() {
         // No-op, handled in the suspendAll terminator
     }
 
+    // TODO: Core could use a QuietDownListener class with `onQuietDown` and `onCancelQuietDown` methods.
+    @Restricted(NoExternalUse.class)
+    @Extension
+    public static class QuietDownListener extends PeriodicWork {
+        private boolean wasQuietDown = Jenkins.get().isQuietingDown();
+
+        @Override
+        public long getRecurrencePeriod() {
+            return Main.isUnitTest
+                    ? TimeUnit.SECONDS.toMillis(1)
+                    : TimeUnit.MINUTES.toMillis(1);
```

Comment:
Not sure about timing here. One minute seems like a decent balance between responsiveness and avoiding unnecessary work, but I am happy to tweak it based on review feedback. Ideally, we'd have a listener so this could be event-driven instead.

Reply:
Not sure if we need to bother with a core patch given that this is kind of a corner case and polling should work well enough.